pkg/ext/stores/tokens/tokens.go

```diff
@@ -375,15 +435,15 @@ func (t *SystemStore) Create(ctx context.Context, group schema.GroupResource, to
 		t.initialized = true
 	}
 
-	ensureNameOrGenerateName(token)
-	// we check the Name directly. because of the ensure... we know that
-	// GenerateName is not set. as it squashes the name in that case.
-	if token.Name != "" {
-		// reject creation of a token which already exists
-		currentSecret, err := t.secretCache.Get(TokenNamespace, token.Name)
-		if err == nil && currentSecret != nil {
-			return nil, apierrors.NewAlreadyExists(group, token.Name)
-		}
+	// enforce our choice of required generateName, and forbidden name.
+	if err := ensureNoNameAndGenerateName(token); err != nil {
+		return nil, err
+	}
```

Comment:
I don't think requiring `generateName` is a good UX.
We should use the default prefix e.g. `token-`.

I mentioned this before, IMHO it would make more sense to always use a predefined prefix (we always generate token names for users).